File: db4oj/core/src/com/db4o/internal/callbacks/NullCallbacks.java
/* Copyright (C) 2006   Versant Inc.   http://www.db4o.com */

package com.db4o.internal.callbacks;

import com.db4o.*;
import com.db4o.ext.*;
import com.db4o.internal.*;
import com.db4o.query.*;

public class NullCallbacks implements Callbacks {

	public void queryOnFinished(Transaction transaction, Query query) {
	}

	public void queryOnStarted(Transaction transaction, Query query) {
	}

	public boolean objectCanNew(Transaction transaction, Object obj) {
		return true;
	}

	public boolean objectCanActivate(Transaction transaction, Object obj) {
		return true;
	}
	
	public boolean objectCanUpdate(Transaction transaction, ObjectInfo objectInfo) {
		return true;
	}
	
	public boolean objectCanDelete(Transaction transaction, ObjectInfo objectInfo) {
		return true;
	}
	
	public boolean objectCanDeactivate(Transaction transaction, ObjectInfo objectInfo) {
		return true;
	}
	
	public void objectOnNew(Transaction transaction, ObjectInfo obj) {
	}
	
	public void objectOnActivate(Transaction transaction, ObjectInfo obj) {
	}

	public void objectOnUpdate(Transaction transaction, ObjectInfo obj) {
	}

	public void objectOnDelete(Transaction transaction, ObjectInfo obj) {
	}

	public void objectOnDeactivate(Transaction transaction, ObjectInfo obj) {	
	}

	public void objectOnInstantiate(Transaction transaction, ObjectInfo obj) {
	}

	public void commitOnStarted(Transaction transaction, CallbackObjectInfoCollections objectInfoCollections) {
	}
	
	public void commitOnCompleted(Transaction transaction, CallbackObjectInfoCollections objectInfoCollections, boolean isOwnCommit) {
	}

	public boolean caresAboutCommitting() {
		return false;
	}

	public boolean caresAboutCommitted() {
		return false;
	}

	public void classOnRegistered(ClassMetadata clazz) {
	}

    public boolean caresAboutDeleting() {
        return false;
    }

    public boolean caresAboutDeleted() {
        return false;
    }

	public void closeOnStarted(ObjectContainer container) {
	}

	public void openOnFinished(ObjectContainer container) {
	}
}

File: db4o.net/Db4objects.Db4o/Db4objects.Db4o/Internal/Callbacks/NullCallbacks.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4objects.Db4o;
using Db4objects.Db4o.Ext;
using Db4objects.Db4o.Internal;
using Db4objects.Db4o.Internal.Callbacks;
using Db4objects.Db4o.Query;

namespace Db4objects.Db4o.Internal.Callbacks
{
	public class NullCallbacks : ICallbacks
	{
		public virtual void QueryOnFinished(Transaction transaction, IQuery query)
		{
		}

		public virtual void QueryOnStarted(Transaction transaction, IQuery query)
		{
		}

		public virtual bool ObjectCanNew(Transaction transaction, object obj)
		{
			return true;
		}

		public virtual bool ObjectCanActivate(Transaction transaction, object obj)
		{
			return true;
		}

		public virtual bool ObjectCanUpdate(Transaction transaction, IObjectInfo objectInfo
			)
		{
			return true;
		}

		public virtual bool ObjectCanDelete(Transaction transaction, IObjectInfo objectInfo
			)
		{
			return true;
		}

		public virtual bool ObjectCanDeactivate(Transaction transaction, IObjectInfo objectInfo
			)
		{
			return true;
		}

		public virtual void ObjectOnNew(Transaction transaction, IObjectInfo obj)
		{
		}

		public virtual void ObjectOnActivate(Transaction transaction, IObjectInfo obj)
		{
		}

		public virtual void ObjectOnUpdate(Transaction transaction, IObjectInfo obj)
		{
		}

		public virtual void ObjectOnDelete(Transaction transaction, IObjectInfo obj)
		{
		}

		public virtual void ObjectOnDeactivate(Transaction transaction, IObjectInfo obj)
		{
		}

		public virtual void ObjectOnInstantiate(Transaction transaction, IObjectInfo obj)
		{
		}

		public virtual void CommitOnStarted(Transaction transaction, CallbackObjectInfoCollections
			 objectInfoCollections)
		{
		}

		public virtual void CommitOnCompleted(Transaction transaction, CallbackObjectInfoCollections
			 objectInfoCollections, bool isOwnCommit)
		{
		}

		public virtual bool CaresAboutCommitting()
		{
			return false;
		}

		public virtual bool CaresAboutCommitted()
		{
			return false;
		}

		public virtual void ClassOnRegistered(ClassMetadata clazz)
		{
		}

		public virtual bool CaresAboutDeleting()
		{
			return false;
		}

		public virtual bool CaresAboutDeleted()
		{
			return false;
		}

		public virtual void CloseOnStarted(IObjectContainer container)
		{
		}

		public virtual void OpenOnFinished(IObjectContainer container)
		{
		}
	}
}
